pilot/pkg/security/model/authentication.go

```diff
@@ -205,8 +209,38 @@ func ConstructValidationContext(rootCAFilePath string, subjectAltNames []string)
 	return ret
 }
 
+// ConstructSANWithTrustDomain constructs MatchSubjectAltNames to be used in DefaultValidationContext by merging
+// Exact SAN matcher user has supplied at Gateway and Prefix SAN matcher for trust domain validation
+func ConstructSANWithTrustDomain(in []string, trustDomainAliases []string) []*matcher.StringMatcher {
```

Comment:
This doesn't need to be exported, and if you follow the below comment, it will not need the `in` parameter.